service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -130,6 +126,31 @@ public String getUserEmailFromSam(AuthenticatedUserRequest userRequest)
     }
   }
 
+  /**
+   * Fetch a user-assigned managed identity that was created for a user, with user credentials,
+   * directly from Sam.
+   */
+  public String getOrCreateUserManagedIdentity(
+      AuthenticatedUserRequest userRequest,
+      String subscriptionId,
+      String tenantId,
+      String managedResourceGroupId)
+      throws InterruptedException {
+    AzureApi azureApi = samAzureApi(userRequest.getRequiredToken());
```

Comment:
Since samAzureApi instantiate new instance of AzureApi maybe it makes sense to rename it to "createSamAzureApi"

Reply:
the name samAzureApi is consistent with samGoogleApi and samUserApi, following the same pattern.